From 94c100548c92849077463522df492298af59fc20 Mon Sep 17 00:00:00 2001
From: Alwin Esch <alwin.esch@web.de>
Date: Fri, 18 Jan 2019 01:11:29 +0100
Subject: [PATCH] [addons] add define ATTRIBUTE_FORCEINLINE on addon header

This add a OS independent defination to force a inline on a function.

This is required on some screensaver and visualization addons to have a clean
code with better performance.

This change does not affect the "C" interface and all previous compiled addons
still work.
---
 .../kodi-addon-dev-kit/include/kodi/AddonBase.h    | 14 ++++++++++++++
 .../kodi-addon-dev-kit/include/kodi/versions.h     |  2 +-
 2 files changed, 15 insertions(+), 1 deletion(-)

diff --git a/xbmc/addons/kodi-addon-dev-kit/include/kodi/AddonBase.h b/xbmc/addons/kodi-addon-dev-kit/include/kodi/AddonBase.h
index 42ce2e73fe92..8e45693a16b6 100644
--- a/xbmc/addons/kodi-addon-dev-kit/include/kodi/AddonBase.h
+++ b/xbmc/addons/kodi-addon-dev-kit/include/kodi/AddonBase.h
@@ -43,6 +43,20 @@
   #define ATTRIBUTE_HIDDEN
 #endif
 
+#ifdef _MSC_VER
+    #define ATTRIBUTE_FORCEINLINE __forceinline
+#elif defined(__GNUC__)
+    #define ATTRIBUTE_FORCEINLINE inline __attribute__((__always_inline__))
+#elif defined(__CLANG__)
+    #if __has_attribute(__always_inline__)
+        #define ATTRIBUTE_FORCEINLINE inline __attribute__((__always_inline__))
+    #else
+        #define ATTRIBUTE_FORCEINLINE inline
+    #endif
+#else
+    #define ATTRIBUTE_FORCEINLINE inline
+#endif
+
 #include "versions.h"
 
 namespace kodi { namespace addon { class CAddonBase; }}
diff --git a/xbmc/addons/kodi-addon-dev-kit/include/kodi/versions.h b/xbmc/addons/kodi-addon-dev-kit/include/kodi/versions.h
index 1c706514efdc..06d75e02c5ff 100644
--- a/xbmc/addons/kodi-addon-dev-kit/include/kodi/versions.h
+++ b/xbmc/addons/kodi-addon-dev-kit/include/kodi/versions.h
@@ -30,7 +30,7 @@
  * overview.
  */
 
-#define ADDON_GLOBAL_VERSION_MAIN                     "1.0.13"
+#define ADDON_GLOBAL_VERSION_MAIN                     "1.0.14"
 #define ADDON_GLOBAL_VERSION_MAIN_MIN                 "1.0.12"
 #define ADDON_GLOBAL_VERSION_MAIN_XML_ID              "kodi.binary.global.main"
 #define ADDON_GLOBAL_VERSION_MAIN_DEPENDS             "AddonBase.h" \
